tasks/bond_configuration.yml

```diff
@@ -13,12 +12,28 @@
   vars:
     bond_check: "{{ item | bond_check }}"
 
+- name: Create the nmconnection file for bond devices
+  become: true
+  template:
+    src: "bond_nmconnection.j2"
+    dest: "/etc/NetworkManager/system-connections/{{ item.device }}.nmconnection"
+    mode: "0600"
+  with_items: '{{ interfaces_bond_interfaces }}'
+  when:
+    - interfaces_use_nmconnection
+  register: bond_result
```

Comment:
This will get shadowed by the other variable of the same name

Reply:
If it weren't for this issue I'd suggest merging then refactoring.

Reply:
So I have a proof of concept on two commits that could fix this; eb4x/ansible-role-interfaces@e437cfbdb3356d1486238bc14e9e44708f9a36aa is a clean-up, and eb4x/ansible-role-interfaces@3215b4a0be7b9745df92b8e152aa6696e6c4be88 introduces new complexity after the clean-up :neutral_face:

I could also rebase these in some other order and do a fixup of eb4x/ansible-role-interfaces@37a9b3d0da3273fb53a7d2f700c63f237d72c461